### ISSUE TITLE
[Custom Components] Support Named Slots in svelte

### ISSUE THREAD

**Issue by Col0ring** [gradio-app/gradio#8706]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Custom components can only use `Default Slot` in svelte currently, therefore it cannot provide more customized layout configurations for users. I'm wondering if gradio can provide a component like `gr.Slot` to support `Named Slot`.

**Describe the solution you'd like**  
This is a sample to use `gr.Slot`:
app.py
```py
with CustomComponent():
    # default slot
    gr.Button()
    # named slot
    with gr.Slot(name="extra"):
        gr.Button()
```

custom_components/Index.svelte
```svelte
<div><slot/></div>
<div><slot name="extra"/></div>
```

**Additional context**

**Comment by abidlabs** [comment]
cc @pngwn

**Comment by pngwn** [comment]
Very interesting. I'll have to think about this a little.

**Comment by pngwn** [comment]
On reflection, I think it might be better to match the Svelte API for this where the slot name is a parameter on the thing being slotted:

```py
with CustomComponent():
    gr.Button() # default slot
    gr.Button(slot="extra") # named slot
```

We could potentially use this mechanism ourselves for things like icons etc in the future. Imagine this:

```py
with gr.Blocks() as demo:
  with gr.Button() as btn:
    gr.Markdown("**Important button**")
    gr.icons.Warning(slot="icon") 
```

**Comment by pngwn** [comment]
We could even expose html primitives for quick bits of UI, food for thought.

**Comment by Col0ring** [comment]
> On reflection, I think it might be better to match the Svelte API for this where the slot name is a parameter on the thing being slotted:
> 
> ```python
> with CustomComponent():
>     gr.Button() # default slot
>     gr.Button(slot="extra") # named slot
> ```
> 
> We could potentially use this mechanism ourselves for things like icons etc in the future. Imagine this:
> 
> ```python
> with gr.IconButton() as btn:
>     gr.Markdown("**Important button**")
>     gr.icons.Warning(slot="icon") 
> ```

LGTM.  One last question, it looks like we can't add multiple components to a named slot  in this way.

```py
with CustomComponent():
    gr.Button() # default slot
    # expected  to be rendered in order in <slot name="extra" />
    gr.Button(slot="extra") # named slot
    gr.Button(slot="extra") # named slot
```

Similar usage to `gr.Slot`, I think it might still be useful to provide an additional component `gr.Fragment` like `svelte:fragment` to resolve this.

```py
with CustomComponent():
    gr.Button() # default slot
    # expected  to be rendered in order in <slot name="extra" />
    with gr.Fragment(slot="extra"): # named slot
       # default slot
       gr.Button() 
       gr.Button() 
```

**Comment by pngwn** [comment]
Yeah maybe, I guess that would reduce the API surface area a little an provide a more explicit API.

**Comment by abidlabs** [comment]
This is out of scope for us unfortunately, as the team is now focused on preparing for Gradio 7. If a workaround is needed, I would suggest creating a completely custom frontend using `gr.Server`: https://huggingface.co/blog/introducing-gradio-server
